### ISSUE TITLE
NaN occurs when building with GaussianNoise, GaussianNoise and Dense

### ISSUE THREAD

**Issue by bugreporter450** [tensorflow/tensorflow#44074]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):a GNU/Linux system with Linux kernel 4.15.0 on 1 6-core 3.60GHz Intel Core CPU i7-6850K with 64 GB RAM equipped with a NVIDIA Corporation GP102 GPUs
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary):source
- TensorFlow version (use command below):tensorflow2.1.0-GPU
- Python version:3.6
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
I got the NaN as loss when I tried to train my model with GaussianNoise, GaussianNoise and Dense
**Describe the expected behavior**
The loss should be a number
**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.
```
batch_size = 110
epochs = 128
num_classes = 10
import os
model_name = 'model.h5'
import tensorflow.keras as keras
(x_train, y_train), (x_test, y_test) = keras.datasets.mnist.load_data()
img_rows, img_cols = x_train.shape[1], x_train.shape[2]

x_train = x_train.reshape(x_train.shape[0], img_rows, img_cols, 1)
x_test = x_test.reshape(x_test.shape[0], img_rows, img_cols, 1)
input_shape = (img_rows, img_cols, 1)
x_train = x_train.astype('float32')
x_test = x_test.astype('float32')
x_train /= 255
x_test /= 255
y_train = keras.utils.to_categorical(y_train, num_classes)
y_test = keras.utils.to_categorical(y_test, num_classes)
model = keras.models.Sequential()
model.add(keras.layers.GaussianNoise(stddev=0.7498748441096037))

model.add(keras.layers.Flatten())
model.add(keras.layers.Dense(num_classes, activation='relu'))
model.compile(loss=keras.losses.categorical_crossentropy,optimizer=keras.optimizers.Adadelta(), metrics=['accuracy'])
model.fit(x_train, y_train, batch_size=batch_size, epochs=epochs, verbose=1, validation_data=(x_test, y_test))
model.save(model_name)
```
**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.
```
Epoch 1/128

  110/60000 [..............................] - ETA: 43:48 - loss: 9.9702 - accuracy: 0.0636
  660/60000 [..............................] - ETA: 7:18 - loss: 9.6680 - accuracy: 0.0848 
 1100/60000 [..............................] - ETA: 4:24 - loss: 9.4182 - accuracy: 0.0991
 1540/60000 [..............................] - ETA: 3:10 - loss: nan - accuracy: 0.1019   
 1760/60000 [..............................] - ETA: 2:47 - loss: nan - accuracy: 0.1000
 2200/60000 [>.............................] - ETA: 2:14 - loss: nan - accuracy: 0.0995
 2530/60000 [>.............................] - ETA: 1:58 - loss: nan - accuracy: 0.0992
 3080/60000 [>.............................] - ETA: 1:37 - loss: nan - accuracy: 0.1042
 3630/60000 [>.............................] - ETA: 1:23 - loss: nan - accuracy: 0.1047
```

**Comment by amahendrakar** [comment]
Was able to reproduce the issue with TF v2.3 and TF-nightly. Please find the gist of it [here](https://colab.research.google.com/gist/amahendrakar/518d78ca0ecfb6c09939051d94a751d9/44074.ipynb). Thanks!

**Comment by nikitamaia** [comment]
Hi @bugreporter450, I think this is because you're using `relu` as the activation on the output layer. Try this instead:
`model.add(keras.layers.Dense(num_classes, activation='softmax'))`
